CMakeLists.txt

```diff
@@ -146,6 +146,15 @@ target_include_directories(build_version PRIVATE
 if(MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi /nologo /EHsc /GS /Gd /GR /GF /fp:precise /Zc:wchar_t /Zc:forScope /errorReport:queue")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /FC /d2Zi+ /W4 /wd4127 /wd4800 /wd4996 /wd4351 /wd4100 /wd4204 /wd4324")
+
+   # Co-routines support
+   option(WITH_COROUTINES "build with coroutines support" OFF)
+   if(WITH_COROUTINES)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /await")
```

Comment:
You may also want to consider adding the `/await:heapelide` flag under optimised builds to turn on the heap-allocation elision feature which avoids heap allocations for coroutine frames it can determine has lifetimes that are strictly nested within the caller.